utils/s2n_safety.c

```diff
@@ -134,8 +143,21 @@ int s2n_constant_time_pkcs1_unpad_or_dont(uint8_t * dst, const uint8_t * src, ui
 
     dont_copy |= src[0] ^ 0x00;
     dont_copy |= src[1] ^ 0x02;
-    dont_copy |= start_of_data[-1] ^ 0x00;
 
+/* Since -1 is being used, we need to disable the pointer overflow check for CBMC. */
+#pragma CPROVER check push
+#pragma CPROVER check disable "pointer-overflow"
+    dont_copy |= start_of_data[-1] ^ 0x00;
+#pragma CPROVER check pop
+
+/* This underflows a value of 0 to the maximum value via arithmetic underflow,
+ * so the check for arithmetic overflow/underflow needs to be disabled for CBMC.
+ * Additionally, uint_fast16_t is defined as the fastest available unsigned
+ * integer with 16 bits or greater, and is not guaranteed to be 16 bits long.
+ * To handle this, the conversion overflow check also needs to be enabled. */
+#pragma CPROVER check push
+#pragma CPROVER check disable "conversion"
+#pragma CPROVER check disable "unsigned-overflow"
     for (uint32_t i = 2; i < srclen - expectlen - 1; i++) {
```

Comment:
is the underflow here? couldn't we avoid that by not performing the operation in `case src- expectlen == 0`?

Reply:
Since this is for code with cryptographic importance I don't feel comfortable changing its operation at run time, but I'll open an issue for it!

Reply:
Could you tag the issue here?